snuba/subscriptions/data.py

```diff
@@ -150,16 +151,26 @@ def build_request(
             Language.LEGACY,
         )
         extra_conditions: Sequence[Condition] = []
-        if offset is not None:
-            extra_conditions = [[["ifNull", ["offset", 0]], "<=", offset]]
+
+        # This check is necessary because datasets like the `sessions` dataset does not contain a
+        # column `offset`
+        if (
+            offset is not None
+            and "offset" in dataset.get_default_entity().get_data_model()
+        ):
+            extra_conditions = [[["ifnull", ["offset", 0]], "<=", offset]]
+
+        build_request_dict = {
+            "project": self.project_id,
+            "conditions": [*self.conditions, *extra_conditions],
+            "aggregations": self.aggregations,
+            "from_date": (timestamp - self.time_window).isoformat(),
+            "to_date": timestamp.isoformat(),
+        }
+        if self.organization:
```

Comment:
There is a lot of dataset specific code here. `self.organization` is only relevant to sessions and it should not be in the shared subscriptions code.

I think we should just skip providing this LegacySubscriptionData implementation altogether if we can as we will not have any legacy items for sessions and this needs to be deprecated anyway.

Reply:
So we can't currently because there is still some request validation that would fail. However, @evanh told me that 
1- We are not using `LegacySubscriptionData` in production anyways 
2- He will be removing class entirely soon enough 

So I suggest I could either not add anything at all to the `LegacySubscriptionData` class to support sessions and just remove the request validation, or just leave these changes as is and when @evanh deletes this class, they will be removed anyways

Reply:
Do you know what's involved in turning off that request validation? (at least for this dataset). If it's straightforward enough it may be worth just doing that rather than providing an implementation that won't be used. It may be worth it since it seems we would need to do that eventually anyway.